hasher-matcher-actioner/hmalib/common/configs/actioner.py

```diff
@@ -48,43 +56,81 @@ class WebhookActionPerformer(ActionPerformer):
     headers: str
 
     def perform_action(self, match_message: MatchMessage) -> None:
-        self.call(data=json.dumps(match_message.to_aws()))
-
-    def call(self, data: str) -> Response:
+        parsed_url, parsed_headers = self.url, self.headers
+        for (
+            replacement_str,
+            replacement_func,
+        ) in WEBHOOK_ACTION_PERFORMER_REPLACEMENTS.items():
+            parsed_url = parsed_url.replace(
+                replacement_str, replacement_func(match_message)
+            )
+            parsed_headers = parsed_headers.replace(
+                replacement_str, replacement_func(match_message)
+            )
+        self.call(parsed_url, parsed_headers, data=json.dumps(match_message.to_aws()))
+
+    def call(self, url: str, headers: str, data: str) -> Response:
         raise NotImplementedError()
 
 
 @dataclass
 class WebhookPostActionPerformer(WebhookActionPerformer):
     """Hit an arbitrary endpoint with a POST"""
 
-    def call(self, data: str) -> Response:
+    def call(self, url: str, headers: str, data: str) -> Response:
         logger.info(f"Performing a POST request to URL. {self.url}")
-        return post(self.url, data, headers=json.loads(self.headers))
+        return post(url, data, headers=json.loads(headers))
 
 
 @dataclass
 class WebhookGetActionPerformer(WebhookActionPerformer):
     """Hit an arbitrary endpoint with a GET"""
 
-    def call(self, data: str) -> Response:
+    def call(self, url: str, headers: str, data: str) -> Response:
         logger.info(f"Performing a GET request to URL. {self.url}")
-        return get(self.url, headers=json.loads(self.headers))
+        return get(url, data, headers=json.loads(headers))
 
 
 @dataclass
 class WebhookPutActionPerformer(WebhookActionPerformer):
     """Hit an arbitrary endpoint with a PUT"""
 
-    def call(self, data: str) -> Response:
+    def call(self, url: str, headers: str, data: str) -> Response:
         logger.info(f"Performing a PUT request to URL. {self.url}")
-        return put(self.url, data, headers=json.loads(self.headers))
+        return put(url, data, headers=json.loads(headers))
 
 
 @dataclass
 class WebhookDeleteActionPerformer(WebhookActionPerformer):
     """Hit an arbitrary endpoint with a DELETE"""
 
-    def call(self, data: str) -> Response:
+    def call(self, url: str, headers: str, _data: str) -> Response:
         logger.info(f"Performing a DELETE request to URL. {self.url}")
-        return delete(self.url, headers=json.loads(self.headers))
+        return delete(url, headers=json.loads(headers))
+
+
+if __name__ == "__main__":
```

Comment:
These "local test pieces"  become outdate if we make any changes and then become an unnecessary layer to debug on top of other things. It also causes you to expand imports when it isn't used anywhere else in the file  (e.g. BankedSignal here) 
What I do and what I would prefer instead of this being committed is having another python file (outside of source control) that can just do:
```
# in no_commit.py for example
from hmalib.common.configs.actioner import  WebhookPostActionPerformer
from hmalib.common.messages.match import BankedSignal, MatchMessage

# the test I want to run manually while making changes
```

if you end up having code use use regularly we can then add it to a unit test or even script/shell command.

Reply:
A lot of this is hard to test using the unit test framework. I get that its not super clean but I actually find it quite useful to store example executions along with the code I'm testing.

Having them linted and checked by mypy has helped me catch bugs as well. If you feel strongly I can delete it but I would prefer to keep it